### PR TITLE
Batch mode

### LINKED ISSUE #272: [ NEW PAGE ] Add documentation on Batch mode
### URL, if applicable

_No response_

### Describe the new content you are requesting.

Qiskit Runtime is introducing a new `Batch` class that works similarly as `Session`, but the former is for submitting multiple jobs at once while the latter is for iterative workloads. 

cc @kt474 

### If this new content request is accepted, do you want to write the content?

I can help the team by providing enough information to write the material

### PR DESCRIPTION
**Pull request by beckykd**
closes #272

### CODE REVIEW

**Review by abbycross — APPROVED** [review]

**Comment by beckykd** [comment]
@javabster - @jyu00 suggested changing the "Sessions" section to "Scheduling mode", so we can put sessions, batch, and Fair share scheduling in there.

File: docs/run/sessions.mdx
Comment (jyu00) [inline docs/run/sessions.mdx:9]:
```suggestion
A session allows a collection of jobs to be grouped and jointly scheduled by the Qiskit Runtime service, facilitating iterative use of quantum computers without incurring queuing delays on each iteration. This eliminates artificial delays caused by other users’ jobs running on the same quantum device during the session. If you don't need iterative processing and can send all jobs at once, use the [`Batch`](run-primitives-batch) class instead.
```
You can still send multiple jobs under `Session`.

File: docs/run/run-primitives-batch.mdx
Comment (jyu00) [inline docs/run/run-primitives-batch.mdx:2]: Why is this titled Run _primitives_ in a batch, whereas the other is Run _jobs_ in a session ?

**Review by jyu00 — COMMENTED** [review]
Mostly looks good. We can add more to the batch section when it's more than just a sham class lol

File: docs/run/run-primitives-batch.mdx
Comment (javabster) [inline docs/run/run-primitives-batch.mdx:9]:
```suggestion
Batch mode is ideal for running experiments closely together to avoid device drifts, that is, to maintain device characterization.  All jobs must be provided at the outset. If you want to submit iterative jobs, use [sessions](sessions) instead.
```

File: docs/run/sessions.mdx
Comment (javabster) [inline docs/run/sessions.mdx:137]:
```suggestion
Sessions run _iteratively_. Running iteratively means that any session job submitted within the interactive timeout, also known as interactive time to live (ITTL), is processed immediately. This allows some time for variational algorithms, such as VQE, to perform classical post-processing. To run jobs in batch mode, see [batch](run-primitives-batch).
```
It feels a little odd to introduce the idea of iteration then jump to batching then go back to explaining iteration, do i'd recommend rephrase like this

**Review by javabster — COMMENTED** [review]
Generally looking good to me, I'll let @jyu00  give the final approve before we merge though.

One question I have just for my own curiosity: am I understanding correctly that a user would want to use sessions primarily if they need to do something that requires some classical postprocessing before sending the next job? And a user would want to use batch mode if they don't need to do that kind of post-processing and just want to send the jobs immediately one after the other? so the deciding factor between sessions or batching would be if you need to do classical postprocessing on each job or not?

File: docs/run/run-primitives-batch.mdx
Comment (beckykd) [reply docs/run/run-primitives-batch.mdx:2]: Good question - seems like they should be consistent.  @javabster do you have a preference?

**Comment by jyu00** [comment]
> One question I have just for my own curiosity: am I understanding correctly that a user would want to use sessions primarily if they need to do something that requires some classical postprocessing before sending the next job? And a user would want to use batch mode if they don't need to do that kind of post-processing and just want to send the jobs immediately one after the other? so the deciding factor between sessions or batching would be if you need to do classical postprocessing on each job or not?

Yes, hence Session has a non-zero interactive TTL (i.e. your next job gets priority as long as it's submitted within X mins) and Batch won't.

File: docs/run/run-primitives-batch.mdx
Comment (pandasa123) [inline docs/run/run-primitives-batch.mdx:9]: Is the note about drift necessarily true? If I have a large experiment that takes several hours, I'll still have drift

Comment (beckykd) [reply docs/run/run-primitives-batch.mdx:9]: I'm not sure - it came from the existing Batch content on the Sessions page.  @jyu00 ?

Comment (jyu00) [reply docs/run/run-primitives-batch.mdx:9]: @pandasa123 you're right that the statement isn't always true. I think the intent here is to describe when you want to use Batch vs single jobs vs session. Some reasons to use batch would be (1) you want to avoid delays between jobs (e.g. to avoid drift) and (2) to speed things up using classical parallelism. 

tbh I think the opening paragraph can use a bit of rewording. It starts by saying it's ideal for avoiding drift, which can be true but not always. Then it lists 2 bullets points, which I think are meant to demonstrate the benefits, but "Suitable for batching many jobs together" isn't exactly a benefit. To me the biggest selling point of Batch is the speed up you gain by having multiple jobs run in parallel.

File: docs/run/run-jobs-batch.mdx
Comment (pandasa123) [inline docs/run/run-jobs-batch.mdx:11]: runing --> running

File: docs/run/run-jobs-batch.mdx
Comment (pandasa123) [inline docs/run/run-jobs-batch.mdx:9]: Is it possible to add links to new tutorials that use batch mode? 

Later today, we'll be updating the [CHSH](https://learning.quantum-computing.ibm.com/tutorial/chsh-inequality) and [Grover](https://learning.quantum-computing.ibm.com/tutorial/grovers-algorithm) one to do exactly that

Comment (beckykd) [reply docs/run/run-jobs-batch.mdx:9]: Of course!

Comment (beckykd) [reply docs/run/run-jobs-batch.mdx:9]: done

File: docs/run/_toc.json
Comment (jyu00) [inline docs/run/_toc.json:44]:
```suggestion
      "title": "Execution modes",
```

`Execution modes` is the term we are presenting at summit for batch + session.

File: docs/run/run-jobs-batch.mdx
Comment (jyu00) [inline docs/run/run-jobs-batch.mdx:18]:
```suggestion
![Batch execution diagram.](/images/run/batch.png 'Figure 1: Batch execution')
```

File: docs/run/run-jobs-batch.mdx
Comment (jyu00) [inline docs/run/run-jobs-batch.mdx:28]: Do we want a full example of using `Batch` (e.g. including import statements)? Since unlike `Session`, there isn't a separate page shows the full code. Or can we just rely on the soon-to-come tutorials?

**Review by jyu00 — APPROVED** [review]

File: docs/run/run-jobs-batch.mdx
Comment (beckykd) [reply docs/run/run-jobs-batch.mdx:28]: A full example might be a good idea - @pandasa123 said that the tutorials that I linked to were updated to use Batch today

Comment (beckykd) [reply docs/run/run-jobs-batch.mdx:28]: I don't see Batch in the Grover's one though; and the CHSH one is slightly different and doesn't use the with Batch() format that we have in this file

Comment (beckykd) [reply docs/run/run-jobs-batch.mdx:28]: They've been updated so I just pointed to them.

**Pull request merged**
